### PR TITLE
Oauth auto allow

### PR DESCRIPTION
**Pull request by zipy124**
This addresses issue #2372

### CODE REVIEW

File: backend/uclapi/oauth/views.py [hunk 1]
```diff
@@ -228,9 +226,46 @@ def shibcallback(request):
...
+            # The code will only be valid for 90 seconds after which redis will
+            # just drop it and the process will be invalidated.
+            r.set(code, verification_data_str, ex=90)
+
+            # Now redirect the user back to the app, at long last.
+            # Just in case they've tried to be super clever and host multiple
+            # apps with the same callback URL, we'll provide the client ID
+            # along with the state
+            return redirect(
+                app.callback_url + "?result=allowed&code=" + code +
```
[the remark this answers is not in the document]
Comment (zipy124) [reply backend/uclapi/oauth/views.py:257]: Hound doesn't accept either before or after it seems? great....

File: backend/uclapi/oauth/views.py [hunk 2]
```diff
@@ -228,9 +226,46 @@ def shibcallback(request):
     }
 
     initial_data = json.dumps(page_data, cls=DjangoJSONEncoder)
-    return render(request, 'permissions.html', {
-        'initial_data': initial_data
-    })
+
+    try:
+        token = OAuthToken.objects.get(app=app, user=user)
```
Comment (Feverfew) [inline backend/uclapi/oauth/views.py:231]: AFAICT this is the only line that can a `OAuthToken.DoesNotExist` exception, so the try/catch block should only surround this line.

Comment (zipy124) [reply backend/uclapi/oauth/views.py:231]: That makes a lot of sense, should probably do that in the rest of the file too retroactively at some point.

**Comment by zipy124** [comment]
> 
> 
> LGTM code wise. I'd like to see some unit tests though, I just can't remember off the top of my head if we have infrastructure for automatically testing functions based on shibcallbacks...

Just had a look, it would appear we do have testing for that. I'll look at adding it in now

**Pull request merged**